### PR TITLE
Bump llvm to f9031f00f2c9

### PR DESCRIPTION
**Pull request by mabubakarpurdue**
As title

### CODE REVIEW

File: test/Dialect/TMTensor/bufferize.mlir [hunk 1]
```diff
@@ -41,8 +41,8 @@ func.func @scan_1d_inclusive(%in: tensor<128xi32>, %out: tensor<128xi32>, %acc:
 // CHECK:             %[[OUT_CURRENT_ELEMENT:.*]] = arith.addi %[[OUT_PREV_ELEMENT]], %[[IN_ELEMENT]] : i32
 // CHECK:             tm_tensor.yield %[[OUT_CURRENT_ELEMENT]] : i32
 // CHECK:           }
-// CHECK:           %[[OUT_TENSOR_NEW:.*]] = bufferization.to_tensor %[[OUT_MEMREF_NEW]] : memref<128xi32>
 // CHECK:           %[[ACC_TENSOR_NEW:.*]] = bufferization.to_tensor %[[ACC_MEMREF_NEW]] : memref<i32>
+// CHECK:           %[[OUT_TENSOR_NEW:.*]] = bufferization.to_tensor %[[OUT_MEMREF_NEW]] : memref<128xi32>
```
Comment (sjain-stanford) [inline test/Dialect/TMTensor/bufferize.mlir:45]: Consider using `CHECK-DAG` for the checks that are order invariant.

Comment (mabubakarpurdue) [reply test/Dialect/TMTensor/bufferize.mlir]: Good point. Fixed all instances

File: test/Conversion/TorchToTosa/basic.mlir [hunk 2]
```diff
@@ -801,10 +801,8 @@ func.func @torch.aten.unsqueeze$negative_dim(%arg0: !torch.vtensor<[4,3],si32> )
 
 // CHECK-LABEL:   func.func @torch.aten.contiguous$basic(
 // CHECK-SAME:                                      %[[VAL_0:.*]]: !torch.vtensor<[?,?],f32>) -> !torch.vtensor<[?,?],f32> {
-// CHECK:           %[[VAL_1:.*]] = torch_c.to_builtin_tensor %[[VAL_0]] : !torch.vtensor<[?,?],f32> -> tensor<?x?xf32>
 // CHECK:           %[[VAL_2:.*]] = torch.constant.int 0
-// CHECK:           %[[VAL_3:.*]] = torch_c.from_builtin_tensor %[[VAL_1]] : tensor<?x?xf32> -> !torch.vtensor<[?,?],f32>
-// CHECK:           return %[[VAL_3]] : !torch.vtensor<[?,?],f32>
+// CHECK:           return %[[VAL_0]] : !torch.vtensor<[?,?],f32>
```
Comment (sjain-stanford) [inline test/Conversion/TorchToTosa/basic.mlir:805]: Curious what llvm/mlir change caused this. Is the `aten.contiguous` just getting canonicalized away in TorchToTosa? cc: @sjarus

**Review by sjain-stanford — APPROVED** [review]
LGTM besides a minor question/comment.

File: test/Conversion/TorchToTosa/basic.mlir (hunk 2 above)
Comment (mabubakarpurdue) [reply test/Conversion/TorchToTosa/basic.mlir:805]: It was being canonicalized anyways in TorchToTosa:

```
LogicalResult ConvertAtenOp<AtenContiguousOp>::matchAndRewrite(
    AtenContiguousOp op, OpAdaptor adaptor,
    ConversionPatternRewriter &rewriter) const {

  // Not a tensor type.
  auto selfType = dyn_cast<TensorType>(adaptor.getSelf().getType());
  if (!selfType)
    return rewriter.notifyMatchFailure(
        op, "Only tensor types are currently supported");

  // FIXME: memory_format is not handled.

  rewriter.replaceOp(op, adaptor.getSelf());

  return success();
}
```

The difference in LLVM/MLIR is that the extra to_builtin_tensor and from_builtin_tensor are canonicalized away by the rewriter

**Pull request merged**